File: mobile/src/main/java/org/dwallach/calwatch/PreferencesHelper.java
/*
 * CalWatch
 * Copyright (C) 2014 by Dan Wallach
 * Home page: http://www.cs.rice.edu/~dwallach/calwatch/
 * Licensing: http://www.cs.rice.edu/~dwallach/calwatch/licensing.html
 */
package org.dwallach.calwatch;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

public class PreferencesHelper {
    private final static String TAG = "PreferencesHelper";

    public static void savePreferences(Context context) {
        Log.v(TAG, "savePreferences");
        SharedPreferences prefs = context.getSharedPreferences("org.dwallach.calwatch.prefs", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = prefs.edit();

        ClockState clockState = ClockState.getSingleton();
        if(clockState == null) {
            Log.e(TAG, "no clock state yet, can't save preferences");
            return;
        }

        editor.putInt("faceMode", clockState.getFaceMode());
        editor.putBoolean("showSeconds", clockState.getShowSeconds());
        editor.putBoolean("showDayDate", clockState.getShowDayDate());

        if(!editor.commit())
            Log.e(TAG, "savePreferences commit failed ?!");
    }

    public static void loadPreferences(Context context) {
        Log.v(TAG, "loadPreferences");

        ClockState clockState = ClockState.getSingleton();
        if(clockState == null) {
            Log.e(TAG, "no clock state yet, can't load preferences");
            return;
        }


        SharedPreferences prefs = context.getSharedPreferences("org.dwallach.calwatch.prefs", Context.MODE_PRIVATE);
        int faceMode = prefs.getInt("faceMode", Constants.DefaultWatchFace); // ClockState.FACE_TOOL
        boolean showSeconds = prefs.getBoolean("showSeconds", Constants.DefaultShowSeconds);
        boolean showDayDate = prefs.getBoolean("showDayDate", Constants.DefaultShowDayDate);

        Log.v(TAG, "faceMode: " + faceMode + ", showSeconds: " + showSeconds + ", showDayDate: " + showDayDate);

        clockState.setFaceMode(faceMode);
        clockState.setShowSeconds(showSeconds);
        clockState.setShowDayDate(showDayDate);

        clockState.pingObservers(); // we only need to do this once, versus multiple times when done internally
    }
}

File: mobile/src/main/kotlin/org/dwallach/calwatch/PreferencesHelper.kt
/*
 * CalWatch
 * Copyright (C) 2014 by Dan Wallach
 * Home page: http://www.cs.rice.edu/~dwallach/calwatch/
 * Licensing: http://www.cs.rice.edu/~dwallach/calwatch/licensing.html
 */
package org.dwallach.calwatch

import android.content.Context
import org.jetbrains.anko.*

object PreferencesHelper: AnkoLogger {
    fun savePreferences(context: Context) {
        verbose("savePreferences")

        context.getSharedPreferences(Constants.PrefsKey, Context.MODE_PRIVATE).edit().apply {
            putInt("faceMode", ClockState.faceMode)
            putBoolean("showSeconds", ClockState.showSeconds)
            putBoolean("showDayDate", ClockState.showDayDate)

            if (!commit())
                error("savePreferences commit failed ?!")
        }
    }

    fun loadPreferences(context: Context) {
        verbose("loadPreferences")

        context.getSharedPreferences(Constants.PrefsKey, Context.MODE_PRIVATE).apply {
            val faceMode = getInt("faceMode", Constants.DefaultWatchFace) // ClockState.FACE_TOOL
            val showSeconds = getBoolean("showSeconds", Constants.DefaultShowSeconds)
            val showDayDate = getBoolean("showDayDate", Constants.DefaultShowDayDate)

            verbose { "faceMode: $faceMode, showSeconds: $showSeconds, showDayDate: $showDayDate" }

            ClockState.faceMode = faceMode
            ClockState.showSeconds = showSeconds
            ClockState.showDayDate = showDayDate
        }

        ClockState.pingObservers() // we only need to do this once, versus multiple times when done internally
    }
}
